applications/FSIApplication/python_scripts/fsi_coupling_interface.py

```diff
@@ -50,61 +96,155 @@ def GetInterfaceModelPart(self):
+        # These values will be used below to calculate the interface residual vector
         self.GetValuesFromFatherModelPart(output_variable)
 
         # Save the previously existent RELAXED_DISPLACEMENT in OLD_RELAXED_DISPLACEMENT before doing the update
-        for node in self.GetInterfaceModelPart().Nodes:
-            relaxed_disp = node.GetSolutionStepValue(KratosMultiphysics.RELAXED_DISPLACEMENT)
-            node.SetSolutionStepValue(KratosMultiphysics.OLD_RELAXED_DISPLACEMENT, relaxed_disp)
+        # These values will be used below to calculate the interface residual vector (u^{k})
+        KratosMultiphysics.VariableUtils().CopyVariable(
+            self._relaxed_variable,
+            self._old_relaxed_variable,
+            self.GetInterfaceModelPart().Nodes)
+
+        # Compute the current non-linear iteration interface residual using the output variable
+        # Note that the residual is computed as r^{k+1} = \tilde{u}^{k+1} - u^{k}
+        self._output_variable_residual_vector = self._get_partitioned_fsi_utilities().SetUpInterfaceVector(self.GetInterfaceModelPart())
+        self._get_partitioned_fsi_utilities().ComputeInterfaceResidualVector(
+            self.GetInterfaceModelPart(),
+            self._old_relaxed_variable,
+            output_variable,
+            self._residual_variable,
+            self._output_variable_residual_vector,
+            "nodal",
+            KratosMultiphysics.FSI_INTERFACE_RESIDUAL_NORM)
 
-        # Get the interface residual size
-        residual_size = self._get_partitioned_fsi_utilities().GetInterfaceResidualSize(self.GetInterfaceModelPart())
+        # Return the current interface residual norm
+        return self.GetInterfaceModelPart().ProcessInfo[KratosMultiphysics.FSI_INTERFACE_RESIDUAL_NORM]
 
+    def Update(self):
         # Set and fill the iteration value vector with the previous non-linear iteration relaxed values (u^{k})
-        iteration_value_vector = KratosMultiphysics.Vector(residual_size)
+        iteration_value_vector = self._get_partitioned_fsi_utilities().SetUpInterfaceVector(self.GetInterfaceModelPart())
         self._get_partitioned_fsi_utilities().InitializeInterfaceVector(
             self.GetInterfaceModelPart(),
-            KratosMultiphysics.OLD_RELAXED_DISPLACEMENT,
+            self._old_relaxed_variable,
             iteration_value_vector)
 
-        # Compute the current non-linear iteration interface residual using the output variable
-        # Note that the residual is computed as r^{k+1} = \tilde{u}^{k+1} - u^{k}
-        output_variable_residual_vector = KratosMultiphysics.Vector(residual_size)
-        self._get_partitioned_fsi_utilities().ComputeInterfaceResidualVector(
+        # Compute the convergence accelerator correction
+        self._get_convergence_accelerator().UpdateSolution(self._output_variable_residual_vector, iteration_value_vector)
```

Comment:
is the indentation correct here??

Reply:
I think it is. For each call of the `Update` method we call the convergence accelerator `UpdateSolution` to correct the initial guess (note that if the indentation wouldn't be correct we'll get a runtime error in this case).